### PR TITLE
fix: usage analytics groups

### PR DESCRIPTION
**Pull request by rephus**
Closes: https://github.com/lightdash/lightdash/issues/10038

### Description:
image

Before:

Screenshot from 2024-05-13 10-30-24

After:

Screenshot from 2024-05-13 10-53-06

Screenshot from 2024-05-13 10-53-17
Screenshot from 2024-05-13 10-53-23

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/backend/src/models/AnalyticsModelSql.ts [hunk 1]
```diff
@@ -4,23 +4,25 @@ export const usersInProjectSql = (
...
   LEFT JOIN projects on project_memberships.project_id = projects.project_id
+  LEFT JOIN group_memberships on group_memberships.user_id = users.user_id
+  LEFT JOIN project_group_access on project_group_access.group_uuid = group_memberships.group_uuid
 WHERE 
   emails.is_primary = true 
   AND ( 
       (organization_memberships.role != 'member'
         AND organization_uuid = '${organizationUuid}')
   OR 
-      (projects.project_uuid = project_uuid
```
Comment (rephus) [inline packages/backend/src/models/AnalyticsModelSql.ts:22]: I believe this was redundant.. right ?

File: packages/backend/src/models/AnalyticsModelSql.ts [hunk 2]
```diff
@@ -4,23 +4,25 @@ export const usersInProjectSql = (
...
-    else project_memberships.role end as role
+  COALESCE(project_memberships.role, project_group_access.role, organization_memberships.role) as role
 from users 
   left join emails on emails.user_id = users.user_id
   LEFT JOIN organization_memberships ON users.user_id  =  organization_memberships.user_id
   LEFT JOIN organizations ON organization_memberships.organization_id = organizations.organization_id
   LEFT JOIN project_memberships ON project_memberships.user_id = users.user_id
   LEFT JOIN projects on project_memberships.project_id = projects.project_id
+  LEFT JOIN group_memberships on group_memberships.user_id = users.user_id
+  LEFT JOIN project_group_access on project_group_access.group_uuid = group_memberships.group_uuid
```
Comment (SimonGodefroid) [inline packages/backend/src/models/AnalyticsModelSql.ts:15]:
```suggestion
  LEFT JOIN group_memberships ON group_memberships.user_id = users.user_id
  LEFT JOIN project_group_access ON project_group_access.group_uuid = group_memberships.group_uuid
```

for consistency

Comment (rephus) [reply packages/backend/src/models/AnalyticsModelSql.ts]: Great suggestion, thanks !

File: packages/backend/src/models/AnalyticsModelSql.ts (hunk 1 above)
Comment (almeidabbm) [reply packages/backend/src/models/AnalyticsModelSql.ts:22]: Yeah, `project_uuid` there seems to me like it is the same as `projects.project_uuid` and therefore redundant, now that you added the join to `project_group_access` you had to explicitly state `projects.project_uuid`

**Review by almeidabbm — APPROVED** [review]

**Pull request merged**